aws_lambda_builders/workflows/java_gradle/workflow.py

```diff
@@ -0,0 +1,72 @@
+"""
+Java Gradle Workflow
+"""
+
+import os
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+from .actions import JavaGradleBuildAction
+from .gradle import SubprocessGradle
+from .utils import OSUtils
+
+
+class JavaGradleWorkflow(BaseWorkflow):
+    """
+    A Lambda builder workflow that knows how to build Java projects using Gradle.
+    """
+    NAME = "JavaGradleWorkflow"
+
+    CAPABILITY = Capability(language="java",
+                            dependency_manager="gradle",
+                            application_framework=None)
+
+    INIT_FILE = "lambda-build-init.gradle"
+
+    def __init__(self,
+                 source_dir,
+                 artifacts_dir,
+                 scratch_dir,
+                 manifest_path,
+                 **kwargs):
+        super(JavaGradleWorkflow, self).__init__(source_dir,
+                                                 artifacts_dir,
+                                                 scratch_dir,
+                                                 manifest_path,
+                                                 **kwargs)
+
+        self.os_utils = OSUtils()
+        gradle_exec = self._determine_gradle_exec()
+        subprocess_gradle = SubprocessGradle(gradle_exec, self.os_utils)
+        artifact_mapping = self._resolve_artifact_mapping()
+
+        self.actions = [
+            JavaGradleBuildAction(source_dir,
+                                  artifacts_dir,
+                                  artifact_mapping,
+                                  subprocess_gradle,
+                                  scratch_dir,
+                                  self.os_utils)
+        ]
+
+    def _determine_gradle_exec(self):
```

Comment:
You should follow the validator and path resolver we just [introduced for this](https://github.com/awslabs/aws-lambda-builders/blob/develop/aws_lambda_builders/workflow.py#L29). Your workflow can define what binaries you need in a [dictionary](https://github.com/awslabs/aws-lambda-builders/blob/develop/aws_lambda_builders/workflow.py#L197). The sanitize method will then validate all binaries and set the first one that is validated. Since order matters here, you can use an OrderedDict to force the resolving of gradlew first.